### PR TITLE
Tweak unit testing developer guide for interfaces

### PR DESCRIPTION
**Pull request by jasdel**
Updates the Unit testing developer guide wording, and examples with context.

### CODE REVIEW

File: content/en/docs/unit-testing.md [hunk 1]
```diff
@@ -98,7 +99,8 @@ func TestGetObjectFromS3(t *testing.T) {
 
 	for i, tt := range cases {
 		t.Run(strconv.Itoa(i), func(t *testing.T) {
-			content, err := GetObjectFromS3(tt.client(t), tt.bucket, tt.key)
+            ctx := context.TODO()
```
Comment (skmcgrail) [inline content/en/docs/unit-testing.md:102]: Tabbing seems off.

File: content/en/docs/unit-testing.md [hunk 2]
```diff
@@ -112,6 +114,11 @@ func TestGetObjectFromS3(t *testing.T) {
 
 ## Mocking Paginators
 
+Similar to service clients, paginators can be mocked by defining a Go interface
+for type type. That interface would be used by your application's code. This
```
Comment (skmcgrail) [inline content/en/docs/unit-testing.md:118]: `type type` => `type`?

File: content/en/docs/unit-testing.md [hunk 3]
```diff
@@ -5,11 +5,15 @@ description: "How to mock the AWS SDK for Go V2 when unit testing your applicati
...
-service client operations, paginators, and waiters.
+When using the SDK in your application, you'll want to mock out the SDK for
+your application's unit test. Mocking out the SDK allows your test to be
+focused on what you want to test, not the internals of the SDK.
+
+To support this your application should use Go interfaces instead of the
+concrete service client, paginators, and waiter types, (e.g. `s3.Client`).
+Using a Go interface to specify the methods your application will use, allows
+your application to use patterns such as to dependency injection. Where the
+concrete implementation is passed in, based on the situation, such as testing.
```
Comment (skmcgrail) [inline content/en/docs/unit-testing.md:16]: Curious on the wording and tone change?

Comment (jasdel) [reply content/en/docs/unit-testing.md:16]: Main intention here was to focus on how user should use the SDK within their application. I don' think the original wording emphasized the how an application would need to be structured/designed in order to be able to mock out the application in it.

**Review by skmcgrail — APPROVED** [review]

**Pull request merged**